traceevents/src/main/java/com/tomtom/kotlin/traceevents/Tracer.kt

```diff
@@ -294,6 +297,25 @@ class Tracer private constructor(
             return null
         }
 
+        /**
+         * Get the parameter names for the method.
+         * For performance reasons, make sure this code is normally executed only once per method call. Note that
+         * this method may be called by multiple threads, so it must be thread-safe as well.
+         * Also make sure the parameter names are only provided if there is exactly 1 name for each parameter only.
+         */
+        val parameterNames: Array<String>?
+        if (parameterNamesCache.containsKey(method)) {
+            parameterNames = parameterNamesCache.get(method)
+        } else {
+            parameterNames = if (args == null) null else {
+                val nonEmptyParameterNames = method.kotlinFunction?.parameters?.mapNotNull { it.name }?.toTypedArray()
+                if (args.size != nonEmptyParameterNames?.size) null else {
+                    parameterNamesCache[method] = nonEmptyParameterNames
+                    nonEmptyParameterNames
+                }
+            }
+        }
```

Comment:
Haven't compiled it, but for idiomatic Kotlin this should be something like:
```suggestion
        val parameterNames: Array<String>? = 
            parameterNamesCache.getOrPut(method) {
                method.kotlinFunction?.parameters
                    ?.mapNotNull { it.name }
                    ?.takeIf { it.size == args?.size }
                    ?.toTypedArray()
            }
```